### PR TITLE
factory: take capabilities setup

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind design
#### What this PR does / why we need it:
moves capabilities setup to the factory package and adds unit tests
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Comment by haircommander** [comment]
@cri-o/cri-o-maintainers PTAL

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by haircommander** [comment]
@saschagrunert :pray:

File: server/container_create.go [hunk 1]
```diff
@@ -286,119 +285,6 @@ func generateUserString(username, imageUser string, uid *types.Int64Value) strin
 	return userstr
 }
 
-// setupCapabilities sets process.capabilities in the OCI runtime config.
-func setupCapabilities(specgen *generate.Generator, caps *types.Capability, defaultCaps capabilities.Capabilities) error {
-	// Remove all ambient capabilities. Kubernetes is not yet ambient capabilities aware
-	// and pods expect that switching to a non-root user results in the capabilities being
-	// dropped. This should be revisited in the future.
-	specgen.Config.Process.Capabilities.Ambient = []string{}
```
Comment (mrunalp) [inline server/container_create.go:294]: This line is missing in the moved function.

Comment (mrunalp) [inline server/container_create.go:294]: This line is missing in the moved function.

File: internal/factory/sandbox/infra.go [hunk 2]
```diff
@@ -48,6 +49,11 @@ func (s *sandbox) InitInfraContainer(serverConfig *libconfig.Config, podContaine
 		return errors.Wrapf(err, "create resolv conf")
 	}
 
+	// Add capabilities from crio.conf if default_capabilities is defined
+	if err := s.infra.SpecSetupCapabilities(&types.Capability{}, serverConfig.DefaultCapabilities); err != nil {
```
Comment (mrunalp) [inline internal/factory/sandbox/infra.go:53]: Do we have an integration test checking this?

Comment (haircommander) [reply internal/factory/sandbox/infra.go:53]: not for pod level. to be honest, it's not totally clear to me why we set capabilities for the infra container

File: server/container_create.go (hunk 1 above)
Comment (haircommander) [reply server/container_create.go:294]: it's done in ClearProcessCapabilities(). as long as it's not updated later we're good

**Comment by haircommander** [comment]
/override ci/prow/e2e-agnostic

(gcp passed)

**Pull request merged**